File: VC类/present/present/ViewController.m
//
//  ViewController.m
//  VCRunTest
//
//  Created by mingyue on 2022/4/11.
//  Copyright © 2022 Gmingyue. All rights reserved.
//

#import "ViewController.h"
#import "ViewControllerA.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
    btn.frame = CGRectMake(0, 0, 100, 30);
    btn.center = self.view.center;
    [btn addTarget:self action:@selector(btnAction) forControlEvents:UIControlEventTouchUpInside];
    [btn setTitle:@"next" forState:UIControlStateNormal];
    [btn setTitleColor:[UIColor redColor] forState:UIControlStateNormal];
    [self.view addSubview:btn];
}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    NSLog(@"%@ %@", NSStringFromClass([self class]), NSStringFromSelector(_cmd));
}

-(void)viewDidAppear:(BOOL)animated
{
    [super viewDidAppear:animated];
}

-(void)viewWillDisappear:(BOOL)animated
{
    [super viewWillDisappear:animated];
}

-(void)viewDidDisappear:(BOOL)animated
{
    [super viewDidDisappear:animated];
}

-(void)btnAction
{
    ViewControllerA * a= [[ViewControllerA alloc] init];
    a.modalPresentationStyle = UIModalPresentationFullScreen;
    [self presentViewController:a animated:YES completion:nil];
    
    NSLog(@"%@-%@", self, self.presentationController);
    NSLog(@"%@-%@", self, self.presentedViewController);
    NSLog(@"%@-%@", self, self.presentingViewController);
}

@end

File: 手势密码/shoushimima/shoushimima/ViewController.swift
//
//  ViewController.swift
//  shoushimima
//
//  Created by mingyue on 15-7-24.
//  Copyright (c) 2015年 csii. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    // MARK: UI elements
    lazy var passwordView: PasswordView = {
        let view = PasswordView(frame: CGRectMake(0, 100, 320, 400))
        view.center.x = self.view.center.x
        view.passwordBlock = { (password) in
            print(password)
        }
        return view
    }()
    
    // MARK: propery
    var passwordStr: String? {
        get {
            return passwordView.passwordString
        }
    }
    
    // MARK: view life circle
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.addSubview(passwordView)
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
    }
    
    // MARK: password methos
    func passwordFunc() {
        guard let passwordStr = passwordStr else {
            return
        }
        print(passwordStr)
    }
}

